packages/eslint-plugin/tests/rules/prefer-find.test.ts

```diff
@@ -584,5 +590,72 @@ arr.find(f, thisArg);
         },
       ],
     },
+    {
```

Comment:
[Testing] Just to be through, could you also add test for a desired report with and within a ternary? Like the `valid` case above, but either with two `.filter`s or with the `[0]` on the other side of the last `)`?

```ts
(Math.random() < 0.5
  ? [1, 2, 3].filter(x => false)
  : [1, 2, 3].filter(x => true))[0];
```

```ts
(Math.random() < 0.5
  ? [1, 2, 3].find(x => true)
  : [1, 2, 3].filter(x => true)[0]);
```

The cases lower down are a bit more complex. It's nice to have more straightforward ones to use as a reference too.